src/Microsoft.VisualStudio.Web.CodeGeneration.Msbuild/ProjectContextWriter.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.ProjectModel;
+using Microsoft.Extensions.ProjectModel.Resolution;
+using Microsoft.VisualStudio.Web.CodeGeneration.Utils;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Msbuild
+{
+    public class ProjectContextWriter : Build.Utilities.Task
```

Comment:
I agree. It is a very round about way of getting the information. We are thinking of moving this in the VS extension direction as you suggested, but that will have to wait for the time being at least.